--- {{cookiecutter.repo_name}}/src/{{cookiecutter.src_package_name}}_fastapi/v1/routers/model.py
+++ {{cookiecutter.repo_name}}/problem-templates/hdb/src/{{cookiecutter.src_package_name}}_fastapi/v1/routers/model.py
@@ -5,0 +6 @@
+import pandas as pd
@@ -18 +19 @@
-def predict(data: str = fastapi.Body()):
+def predict(csv_file: fastapi.UploadFile):
@@ -21,4 +22,4 @@
-    Parameters
-    ----------
-    data : str
-        Input text.
+    Args: 
+    csv_file : fastapi.UploadFile
+        An uploaded CSV file containing the feature columns required by the model.
+            Must match the features used during model training.
@@ -26,4 +27,3 @@
-    Returns
-    -------
-    result_dict : dict
-        Dictionary containing the input string.
+    Returns:
+        FileResponse containing a CSV file with all original columns plus a 
+        prediction' column
@@ -31,5 +31,4 @@
-    Raises
-    ------
-    fastapi.HTTPException
-        A 500 status error is returned if the prediction steps
-        encounters any errors.
+
+    Raises:
+        HTTPException(400): If the uploaded file is not a CSV file.
+        HTTPException(500): If there's an error preventing prediction.
@@ -37 +36,3 @@
-    result_dict = {"data": []}
+    
+    if not csv_file.filename.endswith('.csv'):
+        raise fastapi.HTTPException(400, detail="Only CSV files are supported")
@@ -40 +41,7 @@
-        logger.info("Copying input...")
+        logger.info("Producing prediction")
+        
+        contents = csv_file.file.read()
+        with open(csv_file.filename, "wb") as buffer:
+            buffer.write(contents)
+            
+        dataframe = pd.read_csv(csv_file.filename)
@@ -42 +49,4 @@
-        pred_str = PRED_MODEL.predict(data)
+        predictions = PRED_MODEL.predict(dataframe)
+        dataframe['prediction'] = predictions
+        
+        dataframe.to_csv("./predictions.csv", index=False)
@@ -44,4 +54,5 @@
-        result_dict["data"].append(
-            {"input": pred_str}
-        )
-        logger.info("Input: %s", data)
+        return fastapi.FileResponse(
+                "./predictions.csv",
+                media_type="text/csv",
+                filename="predictions.csv"
+            )
@@ -52,2 +62,0 @@
-
-    return result_dict
