--- {{cookiecutter.repo_name}}/docker/{{cookiecutter.repo_name}}-gpu.Dockerfile
+++ {{cookiecutter.repo_name}}/problem-templates/cv/docker/{{cookiecutter.repo_name}}-gpu.Dockerfile
@@ -35,9 +35,11 @@
 
 # Install pip requirements
 RUN pip install -r {{cookiecutter.repo_name}}/requirements.txt
+RUN pip install -r {{cookiecutter.repo_name}}/pytorch-gpu-requirements.txt
 
 # Use this if deployed outside RunAI
 #RUN micromamba shell init -s bash -p ~/micromamba
 #RUN micromamba install python=3.12.4 -c defaults -n base -y
 #RUN micromamba run -n base pip install -r {{cookiecutter.repo_name}}/requirements.txt
+#RUN micromamba run -n base pip install -r {{cookiecutter.repo_name}}/pytorch-gpu-requirements.txt
 #RUN echo 'alias python="micromamba run -n base python"' >> "${HOME_DIR}/.bashrc"
