--- {{cookiecutter.repo_name}}/src/process_data.py
+++ {{cookiecutter.repo_name}}/problem-templates/hdb/src/process_data.py
@@ -31,20 +31,25 @@
 
     raw_data_dir_path = args["raw_data_dir_path"]
-    dataset = {{cookiecutter.src_package_name_short}}.data_prep.datasets.DummyDataset(
+    dataset = {{cookiecutter.src_package_name_short}}.data_prep.datasets.HdbDataset(
         raw_data_dir_path
     )
+    
+    raw_data = dataset.load_data()
 
     processed_data_dir_path = args["processed_data_dir_path"]
     os.makedirs(args["processed_data_dir_path"], exist_ok=True)
 
-    logger.info(
-        "Copying files from %s to %s.",
-        raw_data_dir_path,
-        processed_data_dir_path
+    ## Example code, adjust or use your own code
+    logger.info("Processing raw data")
+    data_processor = {{cookiecutter.src_package_name_short}}.data_prep.transforms.Transform()
+    X_train, X_test, y_train, y_test = data_processor.transform(
+       raw_data, args["test_size"], args["seed"]
     )
-    dataset.save_data(processed_data_dir_path)
-
-    logger.info("All raw data has been processed.")
-
+    X_train.to_csv(os.path.join(processed_data_dir_path, "X_train.csv"), index=False)
+    X_test.to_csv(os.path.join(processed_data_dir_path, "X_test.csv"), index=False)
+    y_train.to_csv(os.path.join(processed_data_dir_path, "y_train.csv"), index=False)
+    y_test.to_csv(os.path.join(processed_data_dir_path, "y_test.csv"), index=False)
+    logger.info("All raw data has been processed and saved.")
+    ## Example code, adjust or use your own code
 
 if __name__ == "__main__":
