--- {{cookiecutter.repo_name}}/src/train_model.py
+++ {{cookiecutter.repo_name}}/problem-templates/hdb/src/train_model.py
@@ -1,6 +1,7 @@
 """
-This script is for training a dummy model on a dummy dataset.
+This script is for training a XGBRegressor model on the HDB dataset. Note that
+the script is incomplete and requires you to modify it accordingly to your solution.
 """
 import os
 import logging
 import omegaconf
@@ -31,19 +32,31 @@
 
     mlflow_init_status, mlflow_run = {{cookiecutter.src_package_name_short}}.general_utils.mlflow_init(
         args, setup_mlflow=args["setup_mlflow"], autolog=args["mlflow_autolog"]
     )
+    
+    params = {
+        "n_estimators": args["n_estimators"],
+        "learning_rate": args["lr"],
+        "gamma": args["gamma"],
+        "max_depth": args["max_depth"],
+    }
+
+    if use_cuda := args["use_cuda"]:
+       params["device"] = cuda
+       params["tree_method"] = "hist"     
     {{cookiecutter.src_package_name_short}}.general_utils.mlflow_log(
         mlflow_init_status,
         "log_params",
-        params={
-            "dummy_param1": args["dummy_param1"],
-            "dummy_param2": args["dummy_param2"],
-        },
+        params=params,
     )
 
-    model = {{cookiecutter.src_package_name_short}}.modeling.models.DummyModel()
-
+    model = {{cookiecutter.src_package_name_short}}.modeling.models.Model()
+    
+    ### Insert your model code
+    # train_score = model.train(params, X_train, y_train, mlflow_init_status)
+    # test_rmse = model.evaluate()
+    
     {{cookiecutter.src_package_name_short}}.general_utils.mlflow_log(
         mlflow_init_status,
         "log_dict",
         dictionary=omegaconf.OmegaConf.to_container(args, resolve=True),
@@ -51,17 +64,20 @@
     )
 
     os.makedirs(args["artifact_dir_path"], exist_ok=True)
     artifact_path = os.path.join(
-        args["artifact_dir_path"], "output.txt"
+        args["artifact_dir_path"], "xgbreg.json"
     )
+    
+    # model.save_model(os.path.join(args["artifact_dir_path"], "xgbreg.json"))
+    
     with open(artifact_path, "w") as f:
         f.write('\n'.join([f'{x}: {args[x]}' for x in args]))
     {{cookiecutter.src_package_name_short}}.general_utils.mlflow_log(
         mlflow_init_status,
         "log_artifact",
         local_path=artifact_path,
-        artifact_path="outputs"
+        artifact_path="model"
     )
 
     if mlflow_init_status:
         artifact_uri = mlflow.get_artifact_uri()
@@ -76,10 +92,12 @@
         mlflow.end_run()
     else:
         logger.info("Model training has completed.")
 
-    # Outputs for conf/train_model.yaml for hydra.sweeper.direction
-    return args["dummy_param1"], args["dummy_param2"]
+    ## Uncomment return line
+    ## The function decorated with @hydra.main() returns a value that we want to
+    ## adjust base on hydra.sweeper.direction in conf/train_model.yaml
+    # return test_rmse
 
 
 if __name__ == "__main__":
     main()
