--- {{cookiecutter.repo_name}}/src/{{cookiecutter.src_package_name}}_fastapi/config.py
+++ {{cookiecutter.repo_name}}/problem-templates/hdb/src/{{cookiecutter.src_package_name}}_fastapi/config.py
@@ -1,5 +1,6 @@
 """Configuration module for the FastAPI application."""
+
 import pydantic_settings
 
 
 class Settings(pydantic_settings.BaseSettings):
@@ -8,8 +9,10 @@
     API_NAME: str = "{{cookiecutter.project_name}} - Fastapi"
     API_V1_STR: str = "/api/v1"
     LOGGER_CONFIG_PATH: str = "../conf/logging.yaml"
 
-    MODEL_UUID: str = "change-this"
+    USE_CUDA: bool = False
+    PRED_MODEL_UUID: str = "test"
+    PRED_MODEL_PATH: str
 
 
 SETTINGS = Settings()
