--- {{cookiecutter.repo_name}}/src/{{cookiecutter.src_package_name}}_fastapi/config.py
+++ {{cookiecutter.repo_name}}/problem-templates/cv/src/{{cookiecutter.src_package_name}}_fastapi/config.py
@@ -10,5 +10,8 @@
     LOGGER_CONFIG_PATH: str = "../conf/logging.yaml"
 
-    MODEL_UUID: str = "change-this"
+    USE_CUDA: bool = False
+    USE_MPS: bool = False
+    PRED_MODEL_UUID: str = "test"
+    PRED_MODEL_PATH: str
 
 
